18_links_lab/linkslab.txt

```diff
@@ -8,13 +8,15 @@ Preparation: Perform the following  (no need to write the commands you used):
 
   - Create a symbolic link named file3 that points to "file1"
   Answer: 
-
+cd links
+ln -s ./file1 ./file3
   - Create a hard link named file4 that points to "file2"
   Answer: 
-
+ln file2 file4
   - Create a new directory named "tmp" in your home folder:
   Answer:
 
 2. Move "file3" to the new "~/tmp" directory you just created. Can you still use it to see the original "file1" ? Why not? How would you create a symbolic link that would survive the transfer to ~/tmp ?
-
+no, we changed file3 place so the link will not work anymore. we can use "ls -l" which survive thee transfer. 
```

Comment:
Indeed a "hard" link created just with `ln` (without the -s) would survive the transfer, but counts as a hard link.

For a symbolic link to survive the move, it has to point to an absolute path. For example the following code creates a symbolic link that ALWAYS points to /etc/shells, even if moved:

```
ln -s /etc/shells my-link
```

So one option is to use the absolute path of the file for example:

```
ln -s /home/user/links/file1 file3
```

But, if you're already in the directory /home/user/links (because you cd to it earlier), you can use the even shorter version:

```
ln -s $(realpath file1) file3
```

Its meaning: Run the command `realpath file1`, take its output and use it as the argument. Since realpath returns the absolute path to a file it yields the same result as the longer version.